base/threadingconstructs.jl

```diff
@@ -93,31 +93,42 @@ function _threadsfor(iter,lbody)
 end
 
 """
-    Threads.@threads
+    Threads.@threads [schedule] for ... end
 
-A macro to parallelize a for-loop to run with multiple threads. This spawns [`nthreads()`](@ref)
-number of threads, splits the iteration space amongst them, and iterates in parallel.
-A barrier is placed at the end of the loop which waits for all the threads to finish
-execution, and the loop returns.
+A macro to parallelize a `for` loop to run with multiple threads. Splits the iteration
+space among multiple tasks and runs those tasks on threads according to a scheduling
+policy.
+A barrier is placed at the end of the loop which waits for all tasks to finish
+execution.
+
+The `schedule` argument can be used to request a particular scheduling policy.
+The only currently supported value is `static`, which creates one task per thread
+and divides the iterations equally among them.
```

Comment:
This is not actually true, right? It only holds when `Threads.@threads` is used in the main thread. This still works (before and after #35632):

```julia
julia> Threads.@threads for i in 1:Threads.nthreads()
           @assert i == Threads.threadid()
           Threads.@threads for j in 1:Threads.nthreads()
               if i != 1
                   @assert i == Threads.threadid()  # inherits parent tid
               end
           end
       end
```

How about something like

```suggestion
The only currently supported value is `static`, which creates tasks as many as available
threads and divides the iterations equally among them.
```

?

As for "one task per thread" API, I think what people actually need is [`Threads.pin`](https://github.com/JuliaLang/julia/issues/34267#issuecomment-573507670).